hoomd/hpmc/ComputeSDF.h

```diff
@@ -264,23 +269,40 @@ template<class Shape> void ComputeSDF<Shape>::zeroHistogram()
     // Zero the histogram
     for (size_t i = 0; i < m_hist.size(); i++)
         {
-        m_hist[i] = 0;
+        m_hist[i] = 0.0;
         }
     }
 
 /*! \param timestep current timestep
 
     countHistogram() loops through all particle pairs *i,j* where *i* is on the local rank, computes
-    the bin in which that pair should be and adds 1 to the bin. countHistogram() can be called
-    multiple times to increment the counters for averaging, and it operates without any
-    communication.
+    the bin corresponding to the scale factor of the vector r_ij that induces the first overlap
+    between particles *i* and *j*, and then adds s_ij to the bin, where s_ij = 1 - e^{\beta\Delta
+   U_{ij}}. Note that this definition allows for the consideration of soft overlaps where the pair
+   potential changes value, and reduces 1 for hard particle overlaps.  countHistogram() operates
+   without any communication.
       - The integrator performs the ghost exchange (with the ghost width extra that we add)
-      - Only on writeOutput() do we need to sum the per-rank histograms into a global histogram
+
+    This function is a wrapper that calls the appropriate method depending on the value of
+   m_mc->m_patch
 */
 template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
+    {
+    if (m_mc->m_patch)
```

Comment:
```suggestion
    if (m_mc->getPatchEnergy())
```

Reply:
In a future pull request, this will also call the linear search for concave shapes.